api/src/test/java/com/redhat/insights/doubles/DummyTopLevelReport.java

```diff
@@ -41,7 +44,13 @@ public Map<String, Object> getBasic() {
   }
 
   @Override
-  public void decorate(String key, String value) {}
+  public void decorate(String key, String value) {
+    decorations.put(key, value);
+  }
+
+  public Map<String, String> getDecorations() {
+    return decorations;
```

Comment:
this should be an unmodifiable collection. Also this could be package protected

Reply:
Agreed it should be unmodifiable, but it is accessed in runtime tests but declared is api, so it can't be package protected

Reply:
@ehsavoie Unmodifiable collection added.